runtime/gc_realtime/RealtimeAccessBarrier.cpp

```diff
@@ -614,83 +670,6 @@ MM_RealtimeAccessBarrier::checkClassLive(J9JavaVM *javaVM, J9Class *classPtr)
 }
 #endif /* defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING) */
 
-extern "C" {
-
-/**
- * The "barrier" in java.lang.ref.Reference.get() is essentially a complete replacement
- * for the original function.
- */
-J9Object*
-j9gc_objaccess_referenceGet(J9VMThread *vmThread, j9object_t refObject) 
-{
-	MM_RealtimeAccessBarrier *barrier = (MM_RealtimeAccessBarrier*)MM_GCExtensions::getExtensions(vmThread)->accessBarrier;
-	return barrier->referenceGet(vmThread, refObject);
-}
-
-void
-j9gc_objaccess_jniDeleteGlobalReference(J9VMThread *vmThread, J9Object *reference) {
-	MM_RealtimeAccessBarrier *barrier = (MM_RealtimeAccessBarrier*)MM_GCExtensions::getExtensions(vmThread)->accessBarrier;
-	barrier->jniDeleteGlobalReference(vmThread, reference);
-}
-
-/**
- * Check that the two string constants should be considered truly "live".
- * This is a bit of a hack to enable us to scan the string table incrementally.
- * If we are still doing marking work, we treat a call to this method as meaning
- * that one of the strings has been fetched from the string table. If we are finished
- * marking work, but have yet to clear the string table, we treat any unmarked strings
- * as cleared, by returning false from this function.
- * NOTE: Because this function is called from the hash equals function, we can't
- * actually tell which one of the strings is actually the one in the table already,
- * so we have to assume they both are.
- * @return true if they can be considered live, false otherwise.
- */
-UDATA 
-j9gc_objaccess_checkStringConstantsLive(J9JavaVM *javaVM, j9object_t stringOne, j9object_t stringTwo)
```

Comment:
@amicic `j9gc_objaccess_checkStringConstantsLive` returns `UDATA` but equivalent/similar function `j9gc_objaccess_checkStringConstantLive` returns `BOOLEAN`, I moved both "as is" when moving the functions to base class. I=

I think they should be return the same data type, is there any specific reason for the difference, or was it probably just an oversight when being implemented?

Reply:
The difference seems to came from how the callers use them (their return value). Anyhow, they both should use the same type , probably BOOLEAN.

A couple of other related comments/observations.... Did you mean to remove isMetronome check from the caller site?
One of call sites does `checkStringConstantsLive(javaVM, left_s, left_s)`  (note the same arguments) - possibly wrong!?

Reply:
>Did you mean to remove isMetronome check from the caller site?

Ah yes, the check should be removed now. Base implementation is essentially no-op and Metrnome is the only implementer.

>One of call sites does checkStringConstantsLive(javaVM, left_s, left_s) (note the same arguments) - possibly wrong!?

**_Without drilling into the details, It looks like it's intended from the comments:_** 
```C++
/*
* Only consider it to be a match if both string constants are live (i.e. they aren't about to be cleared).
* Since the right string is UTF8, re-use the left string as a placeholder
*/
rc = j9gc_objaccess_checkStringConstantsLive(javaVM, left_s, left_s);
```

We also have this caller that passes in the same param twice:

```C++
/*
* This can only be used if the current candidate pointer is live.
* Pass in candidate twice since we only have one string.
*/
if (FALSE != j9gc_objaccess_checkStringConstantsLive(vm, candidate, candidate)) {
```

Reply:
Another discrepancy between the two apis is where they are declared.


**checkStringConstantLive**
![image](https://user-images.githubusercontent.com/9648788/123355102-f5920280-d532-11eb-889d-eadd1878e53c.png)


**checkStringConstantsLive**
![image](https://user-images.githubusercontent.com/9648788/123355107-f7f45c80-d532-11eb-81be-c5789b9f09bb.png)


These screenshots are a search of the current code base without these changes... 
- Both are defined in `RealtimeAccessBarrier.cpp`
- Both are only used in `StringTable.cpp`
- However, `checkStringConstantsLive` is declared in 2 extra spots: `j9nonbuilder.h` & `gctable.c`

Reply:
There doesn't seem to be a reason for this, should they be simplified, should `checkStringConstantsLive` be removed from j9nonbuilder.h & gctable.c?

Reply:
In terms of return type, I think the difference in return types BOOL vs UDATA between the 2 variants comes from this:


`BOOLEAN
MM_RealtimeAccessBarrier::checkStringConstantLive` will *always* convert `bool` to `BOOLEAN` before returning
```C++
	return _realtimeGC->getMarkingScheme()->isMarked((J9Object *)string) ? TRUE : FALSE;
```

Whereas, `UDATA MM_RealtimeAccessBarrier::checkStringConstantsLive` can return primitive bool: 

```C++
return _realtimeGC->getMarkingScheme()->isMarked((J9Object *)stringOne) && _realtimeGC->getMarkingScheme()->isMarked((J9Object *)stringTwo);
```

Reply:
But according to https://github.com/eclipse-openj9/openj9/pull/12977#discussion_r655583185

UDATA return type should also return `TRUE`/`FALSE` just like `BOOLEAN`.

Reply:
I suppose the only thing to consider is wether `BOOLEAN` can return primitive bool without converting, because that's what `UDATA` has been doing.

Reply:
@amicic fyi

Reply:
They don't seem to be used outside of StringTable, which resides in GC DLL, so no need for inter-module  declaration for either of them. So let's remove the declarations from the one that exists. We could go further and even remove prefix j9gc (why overemphasizing it if it's local to J9/GC code?), but I'm unsure about it (it's still aligns with some all other objaccess API) - up to you if you want to simplify the naming.

I'd also unify the return type and make them both use BOOLEAN (and function should use macros TRUE/FALSE, instead keyword _true/false_). Ideally I'd prefer to use C++ _bool_ (and _true/false_), but BOOLEAN is just still widespread is all other the code involved, so let's stay consistent.

Reply:
Following changes made:
1. Removed the `j9gc` prefix
2. Removed the inter-module declarations 
3. Unified the return type to `BOOLEAN`